velox/tool/QueryTraceToolBase.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <gflags/gflags.h>
+
+DECLARE_bool(usage);
+DECLARE_string(root);
+DECLARE_bool(summary);
+DECLARE_bool(pretty);
+DECLARE_string(task_id);
+
+namespace facebook::velox::tool {
+
+class QueryTraceToolBase {
```

Comment:
```
s/QueryTraceToolBase/QueryTraceReplayer/
```
The tool is built on top of that with environment setup? Like Meta could have it own setup with its internal ecosystem? Thanks!

Reply:
>The tool is built on top of that with environment setup?

Yes, we set up the environment in the `QueryTraceTool::main` like the registering of filesystems, connectors and serder of the plan nodes, vectors, and types.